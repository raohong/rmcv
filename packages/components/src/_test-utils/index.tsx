import { defaultTheme } from '@rmc-vant/system';

export * from '@testing-library/react';
export { default as sleep } from './sleep';

export const getTestThemes = () => defaultTheme;
