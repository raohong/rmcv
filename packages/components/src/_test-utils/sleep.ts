const sleep = (timeout = 0): Promise<void> =>
  new Promise<void>((r) => {
    setTimeout(() => {
      r();
    }, timeout);
  });

export default sleep;
