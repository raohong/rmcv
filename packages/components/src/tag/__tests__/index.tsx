import { Tag } from '..';
import { render } from '../../_test-utils';

it('render correctly', () => {
  const tree = render(<Tag />);

  expect(tree.asFragment()).toMatchSnapshot();
});
