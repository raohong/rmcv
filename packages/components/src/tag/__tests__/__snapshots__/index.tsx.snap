// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0 4px;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  line-height: 16px;
  font-size: 12px;
  color: #fff;
  background: #1989fa;
}

<span
    class="RVuiTag-root RVuiTag-contained RVuiTag-containedPrimary RVuiTag-containedSizeSmall emotion-0"
  />
</DocumentFragment>
`;
